db/seeds/initial_seed.rb

```diff
@@ -7,29 +7,19 @@
 teach_first_cip = CoreInductionProgramme.find_or_create_by!(name: "Teach First")
 ucl_cip = CoreInductionProgramme.find_or_create_by!(name: "UCL Institute of Education")
 
-ambition = LeadProvider.find_or_create_by!(name: "Ambition Institute")
-ambition.update!(cohorts: [cohort_2021]) unless ambition.cohorts.any?
-LeadProviderCip.find_or_create_by!(lead_provider: ambition, cohort: cohort_2021, core_induction_programme: ambition_cip)
-
-bpn = LeadProvider.find_or_create_by!(name: "Best Practice Network")
-bpn.update!(cohorts: [cohort_2021]) unless bpn.cohorts.any?
-LeadProviderCip.find_or_create_by!(lead_provider: bpn, cohort: cohort_2021, core_induction_programme: ucl_cip)
-
-capita = LeadProvider.find_or_create_by!(name: "Capita")
-capita.update!(cohorts: [cohort_2021]) unless capita.cohorts.any?
-LeadProviderCip.find_or_create_by!(lead_provider: capita, cohort: cohort_2021, core_induction_programme: ambition_cip)
-
-edt = LeadProvider.find_or_create_by!(name: "Education Development Trust")
-edt.update!(cohorts: [cohort_2021]) unless edt.cohorts.any?
-LeadProviderCip.find_or_create_by!(lead_provider: edt, cohort: cohort_2021, core_induction_programme: edt_cip)
-
-teach_first = LeadProvider.find_or_create_by!(name: "Teach First")
-teach_first.update!(cohorts: [cohort_2021]) unless teach_first.cohorts.any?
-LeadProviderCip.find_or_create_by!(lead_provider: teach_first, cohort: cohort_2021, core_induction_programme: teach_first_cip)
-
-ucl = LeadProvider.find_or_create_by!(name: "UCL Institute of Education")
-ucl.update!(cohorts: [cohort_2021]) unless ucl.cohorts.any?
-LeadProviderCip.find_or_create_by!(lead_provider: ucl, cohort: cohort_2021, core_induction_programme: ucl_cip)
+[
+  { provider_name: "Ambition Institute", cip: ambition_cip, token: "ambition-token" },
+  { provider_name: "Best Practice Network", cip: ucl_cip, token: "best-practice-token" },
+  { provider_name: "Capita", cip: ambition_cip, token: "capita-token" },
+  { provider_name: "Education Development Trust", cip: edt_cip, token: "edt-token" },
+  { provider_name: "Teach First", cip: teach_first_cip, token: "teach-first-token" },
+  { provider_name: "UCL Institute of Education", cip: ucl_cip, token: "ucl-token" },
+].each do |seed|
+  provider = LeadProvider.find_or_create_by!(name: seed[:provider_name])
+  provider.update!(cohorts: [cohort_2021]) unless provider.cohorts.any?
+  LeadProviderCip.find_or_create_by!(lead_provider: provider, cohort: cohort_2021, core_induction_programme: seed[:cip])
+  LeadProviderApiToken.create_with_known_token!(seed[:token], lead_provider: provider)
+end
```

Comment:
Soo, making this stuff public is not too bad. But I don't want this in production. Can we hide it behind environment check? Or perhaps move token generation to a seed file that doesn't get run in production?

I know we have already ran the seed there, but, well, better safe than sorry.

Reply:
Damn I forgot to move the environment guard across when I moved it in the refactor commit, good spot @kicferk1 , thank you!

Reply:
Done. :sweat_smile: